moped-editor/src/App.js

```diff
@@ -22,42 +23,67 @@ const HASURA_ENDPOINT = process.env.REACT_APP_HASURA_ENDPOINT;
 var pckg = require("../package.json");
 console.info(`🛵 ${pckg.name} ${pckg.version}`);
 
-const App = () => {
-  const restrictedRoutes = restrictRoutes(routes);
-  const routing = useRoutes(restrictedRoutes);
-  const { user } = useUser();
-
-  // https://www.apollographql.com/docs/react/networking/authentication/#header
-  const httpLink = createHttpLink({ uri: HASURA_ENDPOINT });
+const useClient = (user) =>
+  useMemo(() => {
+    // see: https://www.apollographql.com/docs/react/networking/authentication/#header
+    const httpLink = createHttpLink({ uri: HASURA_ENDPOINT });
 
-  const authLink = setContext((_, { headers }) => {
-    // Get the authentication token and role from user if it exists
-    const token = getJwt(user);
-    const role = getHighestRole(user);
+    const authLink = setContext((_, { headers }) => {
+      // Get the authentication token and role from user if it exists
+      const token = getJwt(user);
+      const role = getHighestRole(user);
 
-    // Return the headers and role to the context so httpLink can read them
-    return {
-      headers: {
-        ...headers,
-        authorization: token ? `Bearer ${token}` : "",
-        "x-hasura-role": role ? role : "",
-      },
-    };
-  });
+      // Return the headers and role to the context so httpLink can read them
+      return {
+        headers: {
+          ...headers,
+          authorization: token ? `Bearer ${token}` : "",
+          "x-hasura-role": role ? role : "",
+        },
+      };
+    });
 
-  const client = new ApolloClient({
-    // Join authLink and httpLink to handle auth in each request
-    link: authLink.concat(httpLink),
-    cache: new InMemoryCache(),
-  });
+    return new ApolloClient({
+      // Join authLink and httpLink to handle auth in each request
+      link: authLink.concat(httpLink),
+      cache: new InMemoryCache({
+        typePolicies: {
+          // a type policy must be added for any type we want to cache that
+          // does not use a column called `id` as the PK
+          project_list_view: {
+            keyFields: ["project_id"],
+          },
+          moped_entity: {
+            keyFields: ["entity_id"],
+          },
+          moped_phases: {
+            keyFields: ["phase_id"],
+          },
+          moped_types: {
+            keyFields: ["type_id"],
+          }
```

Comment:
besides the `project_list_view`, i added these type policies for demo purposes. it doesn't really matter that we're caching these values until we set the correct `fetchPolicy` for all instances of `useQuery` that consume this data.